cryptoki/src/types.rs

```diff
@@ -53,6 +53,29 @@ impl Date {
         let date = CK_DATE { year, month, day };
         Self { date }
     }
+
+    /// Creates a new, empty `Date` structure
+    ///
+    /// This represents the default value of the attribute (on
+    /// newer implementations of `Cryptoki`).
+    pub fn new_empty() -> Self {
+        let date = CK_DATE {
```

Comment:
It seems (from my local git grep) `CK_DATE` implements Default so maybe this can be shortened to:

```rust
Self { date: Default::default() }
```

?

Reply:
I meant the entire `date` field could be constructed using `Default`:

```rust
    pub fn new_empty() -> Self {
        Self { date: Default::default() } // or: CK_DATE::default()
    }
```

or even if `Date` implemented `Default` (it could since there is only one field that allegedly impls `Default`) we could have:

```rust
    pub fn new_empty() -> Self {
        Self::default()
    }
```

Please decide which one of all these options here reads better :) (this also reminds of of the law of reviews that smaller ones get a lot of nitpicking while enormous PRs are always "LGTM" haha).

Reply:
Ah, yes, I mindlessly replaced something with `Default::default()` without actually checking what I'm doing. I'll add `Default` derivation to `Date`, since even the comment above this method implies this is the correct value.